esphome/components/uart/__init__.py

```diff
@@ -181,6 +200,7 @@ def maybe_empty_debug(value):
             cv.Required(CONF_BAUD_RATE): cv.int_range(min=1),
             cv.Optional(CONF_TX_PIN): pins.internal_gpio_output_pin_schema,
             cv.Optional(CONF_RX_PIN): validate_rx_pin,
+            cv.Optional(CONF_NAME): cv.string_strict,
```

Comment:
`name` could be confusing here, maybe `port` instead?
And you can also use `cv.file_` to ensure it exists at validation time
```suggestion
            cv.Optional(CONF_PORT): cv.All(cv.file_, cv.only_on(PLATFORM_HOST)),
```